### ISSUE TITLE
Wrong search pattern after selection

### ISSUE THREAD

**Issue by kurkale6ka** [mawww/kakoune#868]
`%s|<ret>` and the search pattern is set to `|` even though it was meant as a regex alternation (`a|b|c`) not a character

**Comment by mawww** [comment]
whats wrong with the behaviour here ? seems the regex lib considers `|` == `.` which seems as good as any interpretation.

**Comment by kurkale6ka** [comment]
yes, that's fine but then if you press `n` for instance you're trying to search for `|` vs `.` say

**Comment by mawww** [comment]
I dont get why you consider that a bug, the regex engine treats the `|` regex as meaning `.`, so `n` will search the next char.
